lib/net/peer.js

```diff
@@ -391,6 +420,11 @@ class Peer extends EventEmitter {
    */
 
   initConnect() {
```

Comment:
This code is directly from the `bcoin` codepath and is used to prevent waiting on `socket.once` forever.